packages/plugin/src/Integrations/PaymentGateways/PayPal/Services/PayPalPriceService.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\Services;
+
+use Solspace\Freeform\Form\Form;
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\Fields\PayPalField;
+
+class PayPalPriceService
+{
+    public function getAmount(Form $form, PayPalField $field): int
+    {
+        $amount = $field->getAmount();
+
+        switch ($field->getAmountType()) {
+            case PayPalField::AMOUNT_TYPE_DYNAMIC:
+                $amount = $this->getDynamicAmount($form, $field);
+
+                break;
+
+            case PayPalField::AMOUNT_TYPE_FIXED:
+            default:
+                $amount = $field->getAmount();
+
+                break;
+        }
```

Comment:
The initial `$amount = $field->getAmount();` is a useless call, since it gets overwritten by the same thing in the `default` case.

Can be changed to a `match` expression, looking cleaner and shorter, more readable.

```suggestion
        $amount = match ($field->getAmountType()) {
            PayPalField::AMOUNT_TYPE_DYNAMIC => $this->getDynamicAmount($form, $field),
            default => $field->getAmount(),
        };
```